GitUI/UserControls/FileStatusItem.cs

```diff
@@ -0,0 +1,26 @@
+﻿using GitCommands;
+
+namespace GitUI.UserControls
+{
+    public sealed class FileStatusItem
+    {
+        public FileStatusItem(GitRevision firstRev, GitRevision secondRev, GitItemStatus item)
+        {
+            SecondRevision = secondRev;
+            FirstRevision = firstRev;
+            Item = item;
+        }
```

Comment:
❗  These all can be `null`, and must either must validate and throw, if not allowed, or be defensive below and handle `null`s at call sites

Reply:
Defensive, similar to before when the information (item/revs) was retrieved from several sources. So null must be checked where used, something I have looked for (but missed the internal usage above).
But this could be improved on. Item should be NotNull, need to review the handling first though.
Revisions are set to null in a few situations currently. This can likely be changed for second (selected) revision, but not for first. Generally first is of no interest for New items and irelevant for the first commit. A random commit hash could be used, but then I prefer null.

Reply:
Updating and added annotations

Reply:
Changed this for SelectedRevision too